ethereum/abi.py

```diff
@@ -32,6 +32,7 @@ def __init__(self, full_signature):
                                  " name. Use %s to call %s with types %r"
                                  % (name, sig_item['name'], encode_types))
             sig = name + '(' + ','.join(encode_types) + ')'
+            sig = sig.encode('utf-8')
```

Comment:
in order to not introduce unicode in the first place, this should better be:
`sig = name + b'(' + b','.join(encode_types) + b')'`